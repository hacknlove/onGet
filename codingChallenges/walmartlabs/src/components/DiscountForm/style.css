.discount-form {
  padding-top: 10px;
}

.apply-button {
  border: solid 1px black;
  border-radius: 18px;
  height: 2.5em;
  font-weight: bold;
  font-size: 14px;
  margin-left: 10px;
}

.apply-button:focus {
  outline: none;
}

#promoCode {
  height: 2.5em;
  border: solid 1px #e6e7e8;
  border-radius: 5px;
  color: gray;
}
#promoCode:focus {
  outline: none;
}

.promoCode-label {
  color: gray;
}