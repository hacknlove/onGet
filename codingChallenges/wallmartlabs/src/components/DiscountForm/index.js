import React, { PureComponent } from 'react';
import './style.css';
import { get, set } from 'onget'

export default class DiscountForm extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      promoCode: ''
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();
    if (this.state.promoCode !== 'DISCOUNT') {
      return
    }
    const pricing = get(`${this.props.url}.pricing`)

    const newDiscount = Math.trunc(pricing.subtotal * 10) / 100;

    set(`${this.props.url}.pricing.total`, pricing.subtotal - pricing.savings - newDiscount + pricing.tax)
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    return (
      <form className="discount-form" onSubmit={this.handleSubmit}>
        <div>
          <label htmlFor="promoCode" className="promoCode-label">
            Promo
          </label>
        </div>
        <div>
          <input
            type="text"
            id="promoCode"
            name="promoCode"
            onChange={this.handleChange}
            value={this.state.promoCode}
          />
          <button className="apply-button">Apply</button>
        </div>
      </form>
    );
  }
}
