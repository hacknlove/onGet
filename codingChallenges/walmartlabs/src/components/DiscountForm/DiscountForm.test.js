import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import DiscountForm from './index';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('discount form component', () => {
  it('matches snapshot', function() {
    let wrapper = shallow(<DiscountForm />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('renders without crashing', function() {
    let wrapper = mount(<DiscountForm />);
    console.log(wrapper.debug());
  });

  it('the input value changes as state changes', function() {
    let wrapper = mount(<DiscountForm />);
    wrapper.setState({ promoCode: 'DISCOUNT' });
    const input = wrapper.find('#promoCode');

    expect(input.matchesElement(<input value="DISCOUNT" />));
  });

  it('the on change the state changes', function() {
    let wrapper = mount(<DiscountForm />);
    const event = { target: { name: 'promoCode', value: 'DISCOUNT' } };
    const input = wrapper.find('#promoCode');
    input.simulate('change', event);

    expect(wrapper.state().promoCode).toBe('DISCOUNT');
  });
});
