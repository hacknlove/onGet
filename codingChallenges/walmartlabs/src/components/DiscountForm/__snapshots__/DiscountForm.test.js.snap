// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`discount form component matches snapshot 1`] = `
<form
  className="discount-form"
  onSubmit={[Function]}
>
  <div>
    <label
      className="promoCode-label"
      htmlFor="promoCode"
    >
      Promo
    </label>
  </div>
  <div>
    <input
      id="promoCode"
      name="promoCode"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="apply-button"
    >
      Apply
    </button>
  </div>
</form>
`;
